broadinstitute/wfl
# [GH-1681] SG/GDC is reprocessing CRAMs and running into Clio update conflicts.

### Purpose


- https://broadinstitute.atlassian.net/browse/GH-1681

### Changes


- Retry add-bam after incrementing the version when Clio suggests force=true.
- Add tests for the new retry logic.
- More with-redefs-fn -=> with-redefs simplifications with some privatization.

### Review Instructions


- Start with the new test in sg_test.clj.
- Check out the implementation in sg.clj.
- I had to add some code for this one.
- The `test-workload-sink-outputs-to-tdr` test still fails
   with `"Load 'loadTagToMonitor' is locked by flight 'x3x8fymvQK-HKzi3XjAo0A'"`
